What is a Nonlinear Solver? And How to easily build Newer Ones (#345)
SciML/NonlinearSolve.jl: docs/src/solvers/steady_state_solvers.md

no algorithm is given, a default algorithm will be chosen.

## Recommended Methods

Conversion to a NonlinearProblem is generally the fastest method. However, this will not
guarantee the preferred root (the stable equilibrium), and thus if the preferred root is
required, then it's recommended that one uses [`DynamicSS`](@ref). For [`DynamicSS`](@ref),
often an adaptive stiff solver, like a Rosenbrock or BDF method (`Rodas5` or `QNDF`), is a
good way to allow for very large time steps as the steady state approaches.

The SteadyStateDiffEq.jl methods on a [`SteadyStateProblem`](@ref) respect the time
definition in the nonlinear definition, i.e., `u' = f(u, t)` uses the correct values for
`t` as the solution evolves. A conversion of a [`SteadyStateProblem`](@ref) to a
[`NonlinearProblem`](@ref) replaces this with the nonlinear system `u' = f(u, ∞)`, and thus
the direct [`SteadyStateProblem`](@ref) approach can give different answers (i.e., the
correct unique fixed point) on ODEs with non-autonomous dynamics.

If you have an unstable equilibrium and you want to solve for the unstable equilibrium,

> then [`DynamicSS`](@ref) might converge to the equilibrium based on the initial condition.

Reviewer: ```suggestion
then [`DynamicSS`](@ref) will not converge to that equilibrium for any initial condition.
```

Author: Done
